nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/AbstractHive3QLProcessor.java

```diff
@@ -345,4 +348,22 @@ private void findTableNames(final Object obj, final Set<TableName> tableNames) {
         }
         return attributes;
     }
+
+    /**
+     * Method to set the configured timeout on the statement to be executed
+     * @param stmt statement to be executed
+     * @param context process context to retrieve the configured value
+     * @param flowFile flow file to evaluate expression language
+     * @throws ProcessException exception in case configured value cannot be converted to an integer
+     */
+    protected void setTimeout(Statement stmt, ProcessContext context, FlowFile flowFile) throws ProcessException {
```

Comment:
Should we have a customValidate() for this like you added for AbstractHiveQLProcessor?

Reply:
Yeah... decided not to because I figured that all the drivers on Hive3 and beyond would implement the method now. I thought it would make less validation work when the framework is validating the processor. But I guess we could choose to be on the safest side and add it anyway. Thoughts?

Reply:
I mentioned it only for consistency between the two sets of processors, actually I'd be fine with not even putting the setQueryTimeout in a try/catch for the Hive 3 processors, now that it's supported, I don't imagine there are any drivers that wouldn't support it at this point.
Either way you want to go is fine with me, let me know what you end up with (even if it is what you already have :) and I'll finish the review and merge, thanks!